src/Peering/Peering/PeeringService/NewAzurePeeringServiceCommand.cs

```diff
@@ -0,0 +1,143 @@
+﻿// ----------------------------------------------------------------------------------
+//
+// Copyright Microsoft Corporation
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+namespace Microsoft.Azure.PowerShell.Cmdlets.Peering.Peering
+{
+    using System;
+    using System.Collections;
+    using System.Linq;
+    using System.Management.Automation;
+
+    using Microsoft.Azure.Commands.Peering.Properties;
+    using Microsoft.Azure.Commands.ResourceManager.Common.ArgumentCompleters;
+    using Microsoft.Azure.Commands.ResourceManager.Common.Tags;
+    using Microsoft.Azure.Management.Peering;
+    using Microsoft.Azure.Management.Peering.Models;
+    using Microsoft.Azure.PowerShell.Cmdlets.Peering.Common;
+    using Microsoft.Azure.PowerShell.Cmdlets.Peering.Models;
+
+    /// <summary>
+    /// New Azure InputObject Command-let
+    /// </summary>
+    [Cmdlet(VerbsCommon.New, "AzPeeringService", DefaultParameterSetName = Constants.ParameterSetNameDefault, SupportsShouldProcess = true)]
+    [OutputType(typeof(PSPeering))]
+    public class NewAzurePeeringServiceCommand : PeeringBaseCmdlet
+    {
+        /// <summary>
+        /// Gets or sets The Resource Group Name
+        /// </summary>
+        [Parameter(
+            Position = 0,
+            Mandatory = true,
+            HelpMessage = Constants.ResourceGroupNameHelp,
+            ParameterSetName = Constants.ParameterSetNameDefault)]
+        [ResourceGroupCompleter]
+        [ValidateNotNullOrEmpty]
+        public string ResourceGroupName { get; set; }
+
+        /// <summary>
+        /// Gets or sets The InputObject NameMD5AuthenticationKeyHelp
+        /// </summary>
+        [Parameter(
+            Position = 1,
+            Mandatory = true,
+            HelpMessage = Constants.PeeringNameHelp,
+            ParameterSetName = Constants.ParameterSetNameDefault)]
+        [ValidateNotNullOrEmpty]
+        public string Name { get; set; }
+
+        /// <summary>
+        /// Gets or sets The InputObject Location.
+        /// </summary>
+        [Parameter(
+            Position = 2,
+            Mandatory = true,
+            HelpMessage = Constants.PeeringServiceLocationHelp,
+            ParameterSetName = Constants.ParameterSetNameDefault)]
+        [ValidateNotNullOrEmpty]
```

Comment:
@rileymckenna please add the `LocationCompleter` attribute to this parameter

Reply:
This is different from an Azure Region. Remember this is one of those One off situations. Ill change the name to -PeeringLocation